extensions/omni_ext/tests/tests.yml

```diff
@@ -0,0 +1,130 @@
+instances:
+  with_omni_ext:
+    default: true
+    config:
+      shared_preload_libraries: */env/OMNI_EXT_SO
+    init:
+    - create extension omni_ext cascade
+
+  with_omni_ext_test:
+    config:
+      shared_preload_libraries: */env/OMNI_EXT_SO
+    init:
+      - create extension omni_ext_test cascade
+
+  with_omni_ext_test_no_preload:
+    config:
+      shared_preload_libraries: */env/OMNI_EXT_SO
+    init:
+      - create extension omni_ext_test_no_preload cascade
+
+tests:
+  - name: load
+    query: select omni_ext.load('omni_ext_test')
+    results:
+      - load: 0.1
+
+  - name: load version
+    steps:
+      - query: select omni_ext.load('omni_ext_test', 'unknown')
+        results:
+          - load:
+      - query: select omni_ext.load('omni_ext_test', '0.1')
+        results:
+          - load: 0.1
+
+  - name: shmem
+    instance: with_omni_ext_test
+    steps:
+      - name: Should be initialized since it's preloaded
+        query: select omni_ext_test.alloc_shmem_global()
+        results:
+          - alloc_shmem_global: test
+
+      - name: database-local
+        query: select omni_ext_test.alloc_shmem_database_local()
+        results:
+          - alloc_shmem_database_local: 'testdb 0'
+      
+#CREATE DATABASE another_db;
+#\connect another_db
+#CREATE EXTENSION omni_ext_test CASCADE;
+#NOTICE:  installing required extension "omni_ext"
+#SELECT omni_ext_test.alloc_shmem_database_local();
+# alloc_shmem_database_local 
+#----------------------------
+# testdb 1
+#(1 row)
```

Comment:
@yrashk Was wondering what would be the ideal way to test this - thought of using a different instance but I guess that defeats the test purposes (which is to create another db).

Reply:
Yeah, it's a good find! If we can't find a solution, I guess we can keep this test intact (that will keep the pg_regress test harness, which is not ideal)

Creating another instance here is beyond the point, as we're testing cross-database behavior in a single instance.

Currently, we can't do `create database` because we're wrapping everything into a transaction (`CREATE DATABASE cannot run inside a transaction block` error), and we can't switch databases on the fly (pg_yregress doesn't have support for it.

To address this, we can perhaps add the following functionality:

## Non-transactional tests in pg_yregress

Something like this?

```yaml
- query: create database another_db
   transaction: false
```

### Ability to send queries to other databases

This one is tougher to design for. Syntax-wise this can be as simple as:

```yaml
- query: select omni_ext_test.alloc_shmem_database_local()
   database: another_db
```

But we would need to connect to manage another connection.

---

If the above will take some time, it may be prudent to merge this PR (assuming it's translating old tests correctly) and open an issue tracking this commented-out test.

Reply:
Extracted this to a bounty: https://github.com/omnigres/omnigres/issues/351